server/src/actions/createUser.ts

```diff
@@ -0,0 +1,18 @@
+// Uploads a new document in the Users collection.
+import { doc, setDoc } from '@firebase/firestore'
+import { users } from '../utilities/firebaseInit'
+
+export const createUser = async (displayName: string, userId: string, avatarUrl: string, lat: number, lon: number, geohash: string) => {
+   const newMsgRef = doc(users, userId) 
+   const docData = {
```

Comment:
Consider adding `const docData: **User** = {...}` for type safety (from types/User.ts)